pkg/search/postgres/store.go

```diff
@@ -268,21 +269,15 @@ func (s *genericStore[T, PT]) GetMany(ctx context.Context, identifiers []string)
 
 	q := search.NewQueryBuilder().AddDocIDs(identifiers...).ProtoQuery()
 
-	rows, err := RunGetManyQueryForSchema[T, PT](ctx, s.schema, q, s.db)
+	resultsByID := make(map[string]PT, len(identifiers))
+	err := RunQueryForSchemaFn(ctx, s.schema, q, s.db, func(msg PT) error {
+		resultsByID[s.pkGetter(msg)] = msg
+		return nil
+	})
 	if err != nil {
-		if errors.Is(err, pgx.ErrNoRows) {
```

Comment:
👍 This section doesn't make a lot of sense as to why it was there to begin with.

Reply:
In meantime we upgraded to pgx v5 and changed the way we iterate over rows but maybe it was never required